x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/Monitoring.java

```diff
@@ -263,13 +277,71 @@ public void reload(Settings settingsToLoad) throws Exception {
+
+            map.entrySet().removeIf(Monitoring::isTypedAPMTemplate);
+
             // this template was not migrated to typeless due to the possibility of the old /_monitoring/bulk API being used
             // see {@link org.elasticsearch.xpack.core.monitoring.exporter.MonitoringTemplateUtils#OLD_TEMPLATE_VERSION}
             // however the bulk API is not typed (the type field is for the docs, a field inside the docs) so it's safe to remove this
             // old template and rely on the updated, typeless, .monitoring-alerts-7 template
             map.remove(".monitoring-alerts");
             return map;
         };
+    }
+
+    /**
+     * Returns a list of template IDs (as defined by {@link MonitoringTemplateUtils#TEMPLATE_IDS}) that are not present in the provided
+     * map or don't have at least {@link MonitoringTemplateUtils#LAST_UPDATED_VERSION} version
+     */
+    static List<String> getMissingMonitoringTemplateIds(Map<String, IndexTemplateMetadata> map) {
+        return Arrays.stream(MonitoringTemplateUtils.TEMPLATE_IDS).filter(id -> {
+            IndexTemplateMetadata templateMetadata = map.get(templateName(id));
+            return templateMetadata == null || (templateMetadata.version() != null && templateMetadata.version() < LAST_UPDATED_VERSION);
+        }).collect(Collectors.toList());
+    }
 
+    /**
+     * Creates the monitoring templates with the provided IDs (must be some of {@link MonitoringTemplateUtils#TEMPLATE_IDS}).
+     * Other ids are ignored.
+     */
+    static List<IndexTemplateMetadata> createMonitoringTemplates(List<String> missingTemplateIds) {
+        List<IndexTemplateMetadata> createdTemplates = new ArrayList<>(missingTemplateIds.size());
+        for (String templateId : missingTemplateIds) {
+            try {
+                final String templateName = MonitoringTemplateUtils.templateName(templateId);
+                final String templateSource = MonitoringTemplateUtils.loadTemplate(templateId);
+                try (
+                    XContentParser parser = XContentType.JSON.xContent()
+                        .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, templateSource)
+                ) {
+                    IndexTemplateMetadata updatedTemplate = IndexTemplateMetadata.Builder.fromXContent(parser, templateName);
+                    logger.info("created template [{}] with version [{}]", templateName, MonitoringTemplateUtils.TEMPLATE_VERSION);
+                    createdTemplates.add(updatedTemplate);
+                } catch (IOException e) {
+                    logger.error("unable to create template [" + templateName + "]", e);
+                }
+            } catch (Exception e) {
+                logger.error("unable to create monitoring template", e);
+            }
+        }
+        return createdTemplates;
+    }
+
+    static boolean isTypedAPMTemplate(Map.Entry<String, IndexTemplateMetadata> templateEntry) {
```

Comment:
the APM template(s) are separate from the monitoring templates. Can you move this out of the monitoring code (likely a different PR) ?

Reply:
> Am I missing something? 

nope

> Is there a better place in Elasticsearch to manage them?

not sure ... i guess this makes sense if there is no other place, although it probably needs a comment to the effect.